### PR TITLE
Support "show parts" command

### LINKED ISSUE #930: Support "show parts" command
```
(user@127.0.0.1) [(none)]> show parts
================================================================================================
| partId | leader     | peers                                          | lost |
================================================================================================
| 1 | 127.0.0.1:1000 | 127.0.0.1:1000, 127.0.0.1:1001, 127.0.0.1:1002 | 127.0.0.1:1002            | 
------------------------------------------------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by zhangguoqing**
1. Add the `ListPartsProcessor`  to get all partitions' information(id, leader, peers, losts) of one graph space.
2. The `SHOW PARTS` command can show the results displayed by console, after doing `USE space_name`. 

close #930

Example:

Init and start the nebula graph cluster:
```
(user@172.25.61.1) [(none)]> show hosts;
==============================================================================================
| Ip          | Port  | Status | Leader count | Leader distribution | Partition distribution |
==============================================================================================
| 172.25.61.1 | 44600 | online | 0            | No valid partition  | No valid partition     |
----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44700 | online | 0            | No valid partition  | No valid partition     |
----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44500 | online | 0            | No valid partition  | No valid partition     |
----------------------------------------------------------------------------------------------
```

Create the graph spaces(named myspace) with 9 partitions and 3 replicas:
```
(user@172.25.61.1) [(none)]> CREATE SPACE myspace(partition_num=9, replica_factor=3);
Execution succeeded (Time spent: 5845/6800 us)
```

Show the hosts with myspace:
```
(user@172.25.61.1) [(none)]> show hosts;
==============================================================================================
| Ip          | Port  | Status | Leader count | Leader distribution | Partition distribution |
==============================================================================================
| 172.25.61.1 | 44600 | online | 0            | myspace: 0          | myspace: 9             |
----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44700 | online | 0            | myspace: 0          | myspace: 9             |
----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44500 | online | 0            | myspace: 0          | myspace: 9             |
----------------------------------------------------------------------------------------------
```

Show parts of myspace, before leader election:
```
(user@172.25.61.1) [(none)]> use myspace;
Execution succeeded (Time spent: 1463/2378 us)

(user@172.25.61.1) [myspace]> show parts;
===========================================================================================
| Partition ID | Leader | Peers                                                   | Losts |
===========================================================================================
| 1            |        | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
-------------------------------------------------------------------------------------------
| 2            |        | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
-------------------------------------------------------------------------------------------
| 3            |        | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
-------------------------------------------------------------------------------------------
| 4            |        | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
-------------------------------------------------------------------------------------------
| 5            |        | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
-------------------------------------------------------------------------------------------
| 6            |        | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
-------------------------------------------------------------------------------------------
| 7            |        | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
-------------------------------------------------------------------------------------------
| 8            |        | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
-------------------------------------------------------------------------------------------
| 9            |        | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
-------------------------------------------------------------------------------------------
```

Show parts of myspace, after leader election:
```
(user@172.25.61.1) [myspace]> show parts;
======================================================================================================
| Partition ID | Leader            | Peers                                                   | Losts |
======================================================================================================
| 1            | 172.25.61.1:44500 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 2            | 172.25.61.1:44500 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 3            | 172.25.61.1:44500 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
| 4            | 172.25.61.1:44500 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 5            | 172.25.61.1:44500 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 6            | 172.25.61.1:44500 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
| 7            | 172.25.61.1:44500 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 8            | 172.25.61.1:44500 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 9            | 172.25.61.1:44500 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
```

Stop the storaged 172.25.61.1:44500:
```
(user@172.25.61.1) [myspace]> show hosts;
==============================================================================================
| Ip          | Port  | Status | Leader count | Leader distribution | Partition distribution |
==============================================================================================
| 172.25.61.1 | 44600 | online | 7            | myspace: 7          | myspace: 9             |
----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44700 | online | 2            | myspace: 2          | myspace: 9             |
----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44500 | online | 0            | myspace: 0          | myspace: 9             |
----------------------------------------------------------------------------------------------
Got 3 rows (Time spent: 155542/156820 us)

(user@172.25.61.1) [myspace]> show parts;
======================================================================================================
| Partition ID | Leader            | Peers                                                   | Losts |
======================================================================================================
| 1            | 172.25.61.1:44600 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 2            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 3            | 172.25.61.1:44600 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
| 4            | 172.25.61.1:44600 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 5            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 6            | 172.25.61.1:44700 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
| 7            | 172.25.61.1:44700 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 8            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 9            | 172.25.61.1:44600 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
```

Wait for the expired_threshold_sec(10 * 60) seconds:
```
(user@172.25.61.1) [myspace]> show hosts;
===============================================================================================
| Ip          | Port  | Status  | Leader count | Leader distribution | Partition distribution |
===============================================================================================
| 172.25.61.1 | 44600 | online  | 7            | myspace: 7          | myspace: 9             |
-----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44700 | online  | 2            | myspace: 2          | myspace: 9             |
-----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44500 | offline | 0            | myspace: 0          | myspace: 9             |
-----------------------------------------------------------------------------------------------
Got 3 rows (Time spent: 2928/3980 us)

(user@172.25.61.1) [myspace]> show parts;
==================================================================================================================
| Partition ID | Leader            | Peers                                                   | Losts             |
==================================================================================================================
| 1            | 172.25.61.1:44600 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 2            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 3            | 172.25.61.1:44600 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 4            | 172.25.61.1:44600 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 5            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 6            | 172.25.61.1:44700 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 7            | 172.25.61.1:44700 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 8            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 9            | 172.25.61.1:44600 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
```
Start the storaged 172.25.61.1:44500 again:
```
[root@YZ-25-61-1 nebula_install1]# ./scripts/nebula-storaged.service start
[INFO] Starting nebula-storaged...
[INFO] Done

[root@YZ-25-61-1 nebula_install1]# bin/nebula --addr=172.25.61.1  --port=8000  --u=user --p=password

Welcome to Nebula Graph

(user@172.25.61.1) [(none)]> show hosts;
==============================================================================================
| Ip          | Port  | Status | Leader count | Leader distribution | Partition distribution |
==============================================================================================
| 172.25.61.1 | 44600 | online | 7            | myspace: 7          | myspace: 9             |
----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44700 | online | 2            | myspace: 2          | myspace: 9             |
----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44500 | online | 0            | myspace: 0          | myspace: 9             |
----------------------------------------------------------------------------------------------
Got 3 rows (Time spent: 3776/4860 us)

(user@172.25.61.1) [(none)]> use myspace;
Execution succeeded (Time spent: 1473/2444 us)

(user@172.25.61.1) [myspace]> show parts;
======================================================================================================
| Partition ID | Leader            | Peers                                                   | Losts |
======================================================================================================
| 1            | 172.25.61.1:44600 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 2            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 3            | 172.25.61.1:44600 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
| 4            | 172.25.61.1:44600 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 5            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 6            | 172.25.61.1:44700 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
| 7            | 172.25.61.1:44700 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 8            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 9            | 172.25.61.1:44600 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
```

### CODE REVIEW

**Comment by dangleptr** [comment]
Jenkins go

**Comment by zhangguoqing** [comment]
> That's cool. Could we show the results displayed by console?

of course!

**Comment by zhangguoqing** [comment]
Jenkins go

**Comment by zhangguoqing** [comment]
Jenkins go

**Comment by dangleptr** [comment]
> > That's cool. Could we show the results displayed by console?
> 
> of course!

You could post it in the summary.

**Comment by zhangguoqing** [comment]
1. Add the `ListPartsProcessor`  to get all partitions' information(id, leader, peers, losts) of one graph space.
2. The `SHOW PARTS` command can show the results displayed by console, after doing `USE space_name`.

**Comment by zhangguoqing** [comment]
> > > That's cool. Could we show the results displayed by console?
> > 
> > 
> > of course!
> 
> You could post it in the summary.

Thanks, I have updated the [summary](https://github.com/vesoft-inc/nebula/pull/1086#issue-328288297) of this PR.

**Comment by dangleptr** [comment]
> > > > That's cool. Could we show the results displayed by console?
> > > 
> > > 
> > > of course!
> > 
> > 
> > You could post it in the summary.
> 
> Thanks, I have updated the [summary](https://github.com/vesoft-inc/nebula/pull/1086#issue-328288297) of this PR.

I means you could post the results displayed by console in this summary.

**Comment by zhangguoqing** [comment]
> > > > > That's cool. Could we show the results displayed by console?
> > > > 
> > > > 
> > > > of course!
> > > 
> > > 
> > > You could post it in the summary.
> > 
> > 
> > Thanks, I have updated the [summary](https://github.com/vesoft-inc/nebula/pull/1086#issue-328288297) of this PR.
> 
> I means you could post the results displayed by console in this summary.

Init and start the nebula graph cluster:
```
(user@172.25.61.1) [(none)]> show hosts;
==============================================================================================
| Ip          | Port  | Status | Leader count | Leader distribution | Partition distribution |
==============================================================================================
| 172.25.61.1 | 44600 | online | 0            | No valid partition  | No valid partition     |
----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44700 | online | 0            | No valid partition  | No valid partition     |
----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44500 | online | 0            | No valid partition  | No valid partition     |
----------------------------------------------------------------------------------------------
```

Create the graph spaces(named myspace) with 9 partitions and 3 replicas:
```
(user@172.25.61.1) [(none)]> CREATE SPACE myspace(partition_num=9, replica_factor=3);
Execution succeeded (Time spent: 5845/6800 us)
```

Show the hosts with myspace:
```
(user@172.25.61.1) [(none)]> show hosts;
==============================================================================================
| Ip          | Port  | Status | Leader count | Leader distribution | Partition distribution |
==============================================================================================
| 172.25.61.1 | 44600 | online | 0            | myspace: 0          | myspace: 9             |
----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44700 | online | 0            | myspace: 0          | myspace: 9             |
----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44500 | online | 0            | myspace: 0          | myspace: 9             |
----------------------------------------------------------------------------------------------
```

Show parts of myspace, before leader election:
```
(user@172.25.61.1) [(none)]> use myspace;
Execution succeeded (Time spent: 1463/2378 us)

(user@172.25.61.1) [myspace]> show parts;
===========================================================================================
| Partition ID | Leader | Peers                                                   | Losts |
===========================================================================================
| 1            |        | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
-------------------------------------------------------------------------------------------
| 2            |        | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
-------------------------------------------------------------------------------------------
| 3            |        | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
-------------------------------------------------------------------------------------------
| 4            |        | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
-------------------------------------------------------------------------------------------
| 5            |        | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
-------------------------------------------------------------------------------------------
| 6            |        | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
-------------------------------------------------------------------------------------------
| 7            |        | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
-------------------------------------------------------------------------------------------
| 8            |        | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
-------------------------------------------------------------------------------------------
| 9            |        | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
-------------------------------------------------------------------------------------------
```

Show parts of myspace, after leader election:
```
(user@172.25.61.1) [myspace]> show parts;
======================================================================================================
| Partition ID | Leader            | Peers                                                   | Losts |
======================================================================================================
| 1            | 172.25.61.1:44500 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 2            | 172.25.61.1:44500 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 3            | 172.25.61.1:44500 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
| 4            | 172.25.61.1:44500 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 5            | 172.25.61.1:44500 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 6            | 172.25.61.1:44500 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
| 7            | 172.25.61.1:44500 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 8            | 172.25.61.1:44500 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 9            | 172.25.61.1:44500 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
```

Stop the storaged 172.25.61.1:44500:
```
(user@172.25.61.1) [myspace]> show hosts;
==============================================================================================
| Ip          | Port  | Status | Leader count | Leader distribution | Partition distribution |
==============================================================================================
| 172.25.61.1 | 44600 | online | 7            | myspace: 7          | myspace: 9             |
----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44700 | online | 2            | myspace: 2          | myspace: 9             |
----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44500 | online | 0            | myspace: 0          | myspace: 9             |
----------------------------------------------------------------------------------------------
Got 3 rows (Time spent: 155542/156820 us)

(user@172.25.61.1) [myspace]> show parts;
======================================================================================================
| Partition ID | Leader            | Peers                                                   | Losts |
======================================================================================================
| 1            | 172.25.61.1:44600 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 2            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 3            | 172.25.61.1:44600 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
| 4            | 172.25.61.1:44600 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 5            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 6            | 172.25.61.1:44700 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
| 7            | 172.25.61.1:44700 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 |       |
------------------------------------------------------------------------------------------------------
| 8            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 |       |
------------------------------------------------------------------------------------------------------
| 9            | 172.25.61.1:44600 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 |       |
------------------------------------------------------------------------------------------------------
```

Wait for the expired_threshold_sec(10 * 60) seconds:
```
(user@172.25.61.1) [myspace]> show hosts;
===============================================================================================
| Ip          | Port  | Status  | Leader count | Leader distribution | Partition distribution |
===============================================================================================
| 172.25.61.1 | 44600 | online  | 7            | myspace: 7          | myspace: 9             |
-----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44700 | online  | 2            | myspace: 2          | myspace: 9             |
-----------------------------------------------------------------------------------------------
| 172.25.61.1 | 44500 | offline | 0            | myspace: 0          | myspace: 9             |
-----------------------------------------------------------------------------------------------
Got 3 rows (Time spent: 2928/3980 us)

(user@172.25.61.1) [myspace]> show parts;
==================================================================================================================
| Partition ID | Leader            | Peers                                                   | Losts             |
==================================================================================================================
| 1            | 172.25.61.1:44600 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 2            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 3            | 172.25.61.1:44600 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 4            | 172.25.61.1:44600 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 5            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 6            | 172.25.61.1:44700 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 7            | 172.25.61.1:44700 | 172.25.61.1:44700, 172.25.61.1:44500, 172.25.61.1:44600 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 8            | 172.25.61.1:44600 | 172.25.61.1:44500, 172.25.61.1:44600, 172.25.61.1:44700 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
| 9            | 172.25.61.1:44600 | 172.25.61.1:44600, 172.25.61.1:44700, 172.25.61.1:44500 | 172.25.61.1:44500 |
------------------------------------------------------------------------------------------------------------------
```

File: src/common/network/NetworkUtils.cpp [hunk 1]
```diff
@@ -306,6 +306,20 @@ StatusOr<std::vector<HostAddr>> NetworkUtils::toHosts(const std::string& peersSt
     return hosts;
 }
 
+std::string NetworkUtils::toHosts(const std::vector<HostAddr>& hosts) {
+    std::string hostsString = "";
+    for (auto& host : hosts) {
+        std::string addrStr = network::NetworkUtils::ipFromHostAddr(host);
+        int32_t port = network::NetworkUtils::portFromHostAddr(host);
+        std::string portStr = folly::to<std::string>(port);
+        hostsString += addrStr + ":" + portStr + ", ";
```
Comment (dangleptr) [inline src/common/network/NetworkUtils.cpp]: folly::stringPrintf("%s:%d,", addrStr, port);

File: src/meta/processors/partsMan/ListPartsProcessor.cpp [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+ */
+
+#include "meta/ActiveHostsMan.h"
+#include "meta/processors/partsMan/ListPartsProcessor.h"
+#include "meta/processors/admin/AdminClient.h"
+
+namespace nebula {
+namespace meta {
+
+void ListPartsProcessor::process(const cpp2::ListPartsReq& req) {
```
Comment (critical27) [inline src/meta/processors/partsMan/ListPartsProcessor.cpp]: Do you think we can support list the partition specified in console? If we have to many partitions, it's hard to look for  a single partition.

Comment (dangleptr) [reply src/meta/processors/partsMan/ListPartsProcessor.cpp]: Agreed.  We'd better support to show some part  information specified too.

Comment (zhangguoqing) [reply src/meta/processors/partsMan/ListPartsProcessor.cpp]: The result is sorted by partition ID.

Comment (dangleptr) [reply src/meta/processors/partsMan/ListPartsProcessor.cpp]: Of course,  we could open another pr for it.

Comment (zhangguoqing) [reply src/meta/processors/partsMan/ListPartsProcessor.cpp]: Yep, would it be better to use the `LIMIT` clause?

File: src/common/network/NetworkUtils.cpp (hunk 1 above)
Comment (zhangguoqing) [reply src/common/network/NetworkUtils.cpp]: Done!

**Comment by zhangguoqing** [comment]
Jenkins go!

File: src/common/network/NetworkUtils.cpp [hunk 3]
```diff
@@ -306,6 +306,19 @@ StatusOr<std::vector<HostAddr>> NetworkUtils::toHosts(const std::string& peersSt
     return hosts;
 }
 
+std::string NetworkUtils::toHosts(const std::vector<HostAddr>& hosts) {
```
Comment (liuyu85cn) [inline src/common/network/NetworkUtils.cpp]: concatHosts may be a better func name to explain this?

Comment (zhangguoqing) [reply src/common/network/NetworkUtils.cpp]: Thx. I prefer the `toHosts` function overload.

Comment (darionyaphet) [reply src/common/network/NetworkUtils.cpp]: `concatHosts` +1

Comment (zhangguoqing) [reply src/common/network/NetworkUtils.cpp]: IMO, the **Concat** generally refers to joining the same kind of things and returning the original form. For example, joining multiple strings returns a string. There is a formal shift here(HostAddr **to** String, with the meaning of convert to), not a simple join or concat.

**Comment by umgsai** [comment]
toHosts +1

File: src/common/network/NetworkUtils.cpp (hunk 3 above)
Comment (umgsai) [reply src/common/network/NetworkUtils.cpp]: toHosts +1

**Comment by jude-zhu** [comment]
close #1226

**Comment by jude-zhu** [comment]
subtask of #724

**Pull request merged**